src/Interpreters/QueryNormalizer.cpp

```diff
@@ -257,7 +269,12 @@ void QueryNormalizer::visit(ASTPtr & ast, Data & data)
     else if (auto * node_select = ast->as<ASTSelectQuery>())
         visit(*node_select, ast, data);
     else if (auto * node_param = ast->as<ASTQueryParameter>())
-        throw Exception("Query parameter " + backQuote(node_param->name) + " was not set", ErrorCodes::UNKNOWN_QUERY_PARAMETER);
+    {
+        if (data.is_create_parameterized_view)
```

Comment:
Seems like my proposal in the comment somewhere above about adding a flag into `ASTQueryParameter` called `allow_non_resolved_parameter` (or any other name which means the same) would be a good fit in this place too.

Reply:
As mentioned in above comment, it would create the same crutch.